### ISSUE TITLE
Broadcast onDisconnect() & onNetworkChange() events

### ISSUE THREAD

**Issue by DEADPXLZ** [Berry-Pool/nami-wallet#17]
Callbacks on Disconnect (aka remove from Whitelist) and on switching between Testnet & Mainnet would be very helpful, consider adding these?

**Comment by alessandrokonrad** [comment]
onNetworkChange is a good idea. Gonna add that one.
With onDisconnect I'm not sure, because as soon as you remove the page from the whitelist, in going and outgoing connections to the webpage are blocked. Also I don't think this would be a typical event or where do you see this as useful?

**Comment by DEADPXLZ** [comment]
100% on the onNetworkChange(), thank you very much!

Regarding the onDisconnect() event, it's useful for updating the UI on apps that use the wallet during runtime.
Metamask also does this and they actually go the extra mile where you can actually trigger a disconnection from within the 3rd party website. LMKWYT

**Comment by alessandrokonrad** [comment]
So onDisconnect would probably work best then if an endpoint `cardano.disable()` exists as well right?
I'll think about the onDisconnect event.

**Comment by DEADPXLZ** [comment]
Agreed! Listening for disconnections is important as well as it enables more interaction via 3rd party apps and gets users closer to the classic 'account' user journey.

**Comment by cjkoepke** [comment]
> closer to the classic 'account' user journey.

This is the main point, user's should be able to interface with the wallet without necessarily opening the extension.

**Comment by alessandrokonrad** [comment]
I get the idea, but the reason why you connect the wallet in first place is because you trust the website you visit. Why would you want to disconnect it again?

**Comment by cjkoepke** [comment]
@alessandrokonrad For instance, I am building an interface that uses NFTs within your wallet to "login" to the website. If the user wants to log out (privacy or other), disconnecting your wallet could theoretically act as a catalyst or action to do so.

Also, what if you change your mind about trusting a website?

**Comment by rooooooooob** [comment]
Perhaps your view changed of the website, or you only wanted to check it out but would prefer it not have automatic access from now on, depending on how you're implementing your whitelist. It is probably something good to have. Could we start the discussion in the cardano-foundation/CIPs#88 over which events we need to put in the standard? It was an oversight I think as I had included a disconnect event in the Ergo EIP-0012 standard we based the CIP on.

**Comment by rooooooooob** [comment]
As the initial CIP-30 PR cardano-foundation/CIPs#88 was merged without an events API (since it wasn't finalized), we have put up a new PR cardano-foundation/CIPs#151 to amend it for events. Events-related discussion should take place in that PR.
